boards/arm/disco_l475_iot1/disco_l475_iot1.dts

```diff
@@ -45,6 +45,10 @@
 	};
 };
 
+&soc {
+	compatible = "STM32L475XG", "simple-bus";
```

Comment:
We already have : 
```
compatible = "st,stm32f207zg-nucleo", "st,stm32f207";
```
Can't we re-use that instead?

Reply:
@erwango 

You may have a look at the extended flash property extraction to EDTS in 733ecb0ea4462b8ebd561935349d12beaa17e8e5 of #5799.

Reply:
So I guess I'll go with the direct route of having stm32f207zg.dtsi and including that in the board.dts.

Reply:
I was hoping we could do that in the board and spare the stm32f207zg to avoid multiplication of dtsi file. Can't we ?

Reply:
Like done in linux (https://elixir.bootlin.com/linux/latest/source/arch/arm/boot/dts/stm32f429-disco.dts):
```
	memory {
		reg = <0x90000000 0x800000>;
	};
```

Reply:
I wanted to avoid having boards that use the same SoC having to duplicate that info.  While we could do it in the board.dts I think having a SoC specific .dtsi is the cleanest.